### PR TITLE
Add Reference Links to Decoder Docs

### PR DESCRIPTION
**Pull request by thebombzen**
This PR will hyperlink functions included in the docs so you don't need to search them with Ctrl+F.

### CODE REVIEW

**Comment by thebombzen** [comment]
Helpful link: [https://libjxl--1421.org.readthedocs.build/en/1421/api_decoder.html](Read The Docs PR Build)

**Comment by thebombzen** [comment]
Pending approval, I'll make a similar change to encoder.h and the other files with docs.

File: lib/include/jxl/decode.h [hunk 1]
```diff
@@ -561,9 +569,9 @@ JXL_EXPORT void JxlDecoderCloseInput(JxlDecoder* dec);
  * @param dec decoder object
  * @param info struct to copy the information into, or NULL to only check
  * whether the information is available through the return value.
- * @return JXL_DEC_SUCCESS if the value is available,
- *    JXL_DEC_NEED_MORE_INPUT if not yet available, JXL_DEC_ERROR in case
- *    of other error conditions.
+ * @return @ref JXL_DEC_SUCCESS if the value is available,
+ *    @ref JXL_DEC_NEED_MORE_INPUT if not yet available, @ref JXL_DEC_ERROR in
+ * case of other error conditions.
```
Comment (veluca93) [inline lib/include/jxl/decode.h:574]: Indentation seems a bit messed up here

File: lib/include/jxl/decode.h [hunk 2]
```diff
@@ -130,162 +131,165 @@ typedef enum {
...
-  /** Informative event by JxlDecoderProcessInput: basic information such as
-   * image dimensions and extra channels. This event occurs max once per image.
+  /** Informative event by @ref JxlDecoderProcessInput - Basic information such
+   * as image dimensions and extra channels. This event occurs max once per
+   * image.
    */
   JXL_DEC_BASIC_INFO = 0x40,
 
-  /** Informative event by JxlDecoderProcessInput: user extensions of the
+  /** Informative event by @ref JxlDecoderProcessInput - User extensions of the
```
Comment (veluca93) [inline lib/include/jxl/decode.h:189]: Why the change from : to -?

Comment (thebombzen) [reply lib/include/jxl/decode.h:189]: Docs fail to build otherwise since it grabs the `:` as part of the function name. I'm not sure how to work around that so I didn't.

File: lib/include/jxl/decode.h (hunk 1 above)
Comment (thebombzen) [reply lib/include/jxl/decode.h:574]: Ugh, I blame clang-format. I'll fix this one.

File: lib/include/jxl/decode.h [hunk 3]
```diff
@@ -66,57 +66,58 @@ typedef enum {
...
- *   - JXL_SIG_NOT_ENOUGH_BYTES not enough bytes were passed to determine
- *       if a valid signature is there.
- *   - JXL_SIG_INVALID: no valid signature found for JPEG XL decoding.
- *   - JXL_SIG_CODESTREAM a valid JPEG XL codestream signature was found.
- *   - JXL_SIG_CONTAINER a valid JPEG XL container signature was found.
+ *   - @ref JXL_SIG_NOT_ENOUGH_BYTES if not enough bytes were passed to
+ * determine if a valid signature is there.
+ *   - @ref JXL_SIG_INVALID if no valid signature found for JPEG XL decoding.
+ *   - @ref JXL_SIG_CODESTREAM if a valid JPEG XL codestream signature was
+ * found.
```
Comment (veluca93) [inline lib/include/jxl/decode.h:73]: still a little bit of wonky indentation :)

File: lib/include/jxl/decode.h (hunk 1 above)
Comment (veluca93) [reply lib/include/jxl/decode.h:582]: (also below)

I think I'd prefer sticking with indenting as it was before though.

File: lib/include/jxl/decode.h (hunk 3 above)
Comment (thebombzen) [reply lib/include/jxl/decode.h:73]: I think the extra space is necessary here because of the list items. Let me go scan the rest of it for wonky indentation though.

Comment (veluca93) [reply lib/include/jxl/decode.h:73]: Just TBC, I meant that's not enough indentation ;)

**Review by veluca93 — APPROVED** [review]
